plugins/woocommerce/includes/wc-cart-functions.php

```diff
@@ -179,21 +182,33 @@ function wc_clear_cart_after_payment() {
 			$order = wc_get_order( $order_id );
 
 			if ( $order instanceof WC_Order && hash_equals( $order->get_order_key(), $order_key ) ) {
-				WC()->cart->empty_cart();
+				$should_clear_cart_after_payment = true;
```

Comment:
**(2/4)** Then here, we of course believe a payment just took place so we update the new var.

```suggestion
				$after_payment                   = true;
				$should_clear_cart_after_payment = true;
```